### PR TITLE
Swc 1117

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/entity/EntityPageTopViewImpl.java [hunk 1]
```diff
@@ -806,4 +767,24 @@ private Widget createDescriptionWidget(final EntityBundle bundle, String entityT
...
+	/**
+	 * Create a new action menu for an entity.
+	 * @param bundle
+	 * @return
+	 */
+	public ActionMenuWidget createEntityActionMenu(EntityBundle bundle){
+		// Create a menu
+		ActionMenuWidget actionMenu = ginInjector.createActionMenuWidget();
+		// Create a controller.
+		EntityActionController controller = ginInjector.createEntityActionController();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/EntityPageTopViewImpl.java:779]: EntityActionController could be in ActionMenuWidget

File: src/main/java/org/sagebionetworks/web/client/widget/table/v2/TableModelUtils.java [hunk 2]
```diff
@@ -11,7 +11,7 @@
 import org.sagebionetworks.schema.adapter.AdapterFactory;
 import org.sagebionetworks.schema.adapter.JSONEntity;
 import org.sagebionetworks.schema.adapter.JSONObjectAdapterException;
-import org.sagebionetworks.web.client.model.EntityBundle;
+import org.sagebionetworks.repo.model.EntityBundle;
 import org.sagebionetworks.web.shared.EntityBundleTransport;
 
 import com.google.inject.Inject;
```
Comment (john-hill) [inline src/main/java/org/sagebionetworks/web/client/widget/table/v2/TableModelUtils.java]: remove class

File: src/main/java/org/sagebionetworks/web/client/widget/table/v2/TableModelUtils.java [hunk 3]
```diff
@@ -113,7 +113,10 @@ public TableEntity tableEntityFromJSON(String json) throws JSONObjectAdapterExce
 	public EntityBundle bundleFromTransport(EntityBundleTransport transport) throws JSONObjectAdapterException{
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/table/v2/TableModelUtils.java]: can remove?

File: src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java [hunk 4]
```diff
@@ -430,72 +398,72 @@ private JSONObject query(String query) throws SynapseException {
...
+//						acl = getAcl(entityId);
+//					} catch (SynapseException e) {
+//						e.printStackTrace();
+//					}
+//				}
+//				ebt.setAcl(acl);
+//			}
+//			if ((EntityBundleTransport.ACCESS_REQUIREMENTS & partsMask) != 0) {
+//				ebt.setAccessRequirements(eb.getAccessRequirements());
+//			}
```
Comment (john-hill) [inline src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java]: remove

File: src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java [hunk 5]
```diff
@@ -430,72 +398,72 @@ private JSONObject query(String query) throws SynapseException {
...
-			}
-			
-			ebt.setIsWikiBasedEntity(getWikiBasedEntities().contains(entityId));
-
-		} catch (JSONObjectAdapterException e) {
-			throw new UnknownErrorException(e.getMessage());
-		}
-		return ebt;
-	}
+//	// Convert repo-side EntityBundle to serializable EntityBundleTransport
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java]: remove

File: src/main/java/org/sagebionetworks/web/shared/EntityBundle.java [hunk 6]
```diff
@@ -1,7 +1,8 @@
-package org.sagebionetworks.web.client.model;
+package org.sagebionetworks.web.shared;
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/shared/EntityBundle.java]: delete

File: src/main/java/org/sagebionetworks/web/shared/EntityBundle.java [hunk 7]
```diff
@@ -10,6 +11,8 @@
 import org.sagebionetworks.repo.model.file.FileHandle;
 import org.sagebionetworks.repo.model.table.TableBundle;
 
+import com.google.gwt.user.client.rpc.IsSerializable;
```
Comment (john-hill) [inline src/main/java/org/sagebionetworks/web/shared/EntityBundle.java]: delete

File: src/main/resources/org/sagebionetworks/web/client/widget/entity/menu/v2/ActionMenuWidgetViewImpl.ui.xml [hunk 8]
```diff
@@ -4,28 +4,37 @@
...
+						<a:ActionMenuItem icon="UPLOAD" action="UPLOAD_TABLE_DATA">Upload
+							Data To Table</a:ActionMenuItem>
+						<a:ActionMenuItem action="EDIT_TABLE_DATA">Edit
+							Query Results</a:ActionMenuItem>
+						<a:ActionMenuItem icon="DOWNLOAD"
+							action="DOWNLOAD_TABLE_QUERY_RESULTS">Download Query Results</a:ActionMenuItem>
+						<b:Divider ui:field="basicDivider" visible="false" />
+						<a:ActionMenuItem icon="EDIT" action="CHANGE_ENTITY_NAME">Change Name</a:ActionMenuItem>
+						<a:ActionMenuItem icon="PLUS" action="ADD_WIKI_PAGE">Add Wiki</a:ActionMenuItem>
+						<a:ActionMenuItem action="SUBMIT_TO_CHALLENGE">Submit To Challenge</a:ActionMenuItem>
```
Comment (john-hill) [inline src/main/resources/org/sagebionetworks/web/client/widget/entity/menu/v2/ActionMenuWidgetViewImpl.ui.xml]: trophy icon

Comment (jay-hodgson) [inline src/main/resources/org/sagebionetworks/web/client/widget/entity/menu/v2/ActionMenuWidgetViewImpl.ui.xml]: use Trophy, like challenge items

**Pull request merged**